arch/risc-v/src/common/riscv_addrenv.c

```diff
@@ -81,6 +81,12 @@
 #  error "This module is intended to be used with CONFIG_BUILD_KERNEL"
 #endif
 
+/* Make sure the address environment virtual address boundary is valid */
+
+#if (ADDRENV_VBASE & RV_MMU_SECTION_ALIGN) != 0
+#  error "Addrenv start address is not aligned to section boundary"
+#endif
```

Comment:
```suggestion
static_assert(ADDRENV_VBASE & RV_MMU_SECTION_ALIGN) == 0,
              "Addrenv start address is not aligned to section boundary");
```
How about using the cleaner `static_assert` instead?

Reply:
Fine by me. I selected this because I noticed many other files use

```
#if ZZZ 
#  error "Error happened!"
#endif
```